src/config/mod.rs

```diff
@@ -93,8 +100,8 @@ impl<'c> Config<'c> {
     /// Create boot source from `self`.
     pub fn boot_source(&self) -> BootSource<'_> {
         BootSource {
-            kernel_image_path: self.kernel_image_path.as_ref(),
-            initrd_path: self.initrd_path.as_ref().map(AsRef::as_ref),
+            kernel_image_path: self.kernel_image_file.as_ref(),
+            initrd_path: self.initrd_file.as_ref().map(AsRef::as_ref),
```

Comment:
These are expected to be paths (relative to the chroot), not just filenames so I'm a bit confused about the rationale for these changes here. My bad that I didn't notice in the review but please change it ASAP.